src/haz3lweb/Main.re

```diff
@@ -102,17 +113,37 @@ module App = {
       (
         model: Incr.t(Haz3lweb.Model.t),
         ~old_model as _: Incr.t(Haz3lweb.Model.t),
-        ~inject,
+        ~inject: UpdateAction.t => Ui_effect.t(unit),
       ) => {
     open Incr.Let_syntax;
     let%map model = model;
     /* Note: mapping over the old_model here may
        trigger an additional redraw */
+    let first = ref(false);
+    let go = ref(false);
+    if (!first.contents) {
+      let _eventId =
+        Dom_html.addEventListener(
+          Dom_html.document,
+          Dom_html.Event.make("myCustomEvent"),
+          Dom_html.handler(_ => {
+            print_endline("RECEIVED EVENT");
+            go := true;
+            Js.bool(true);
+          }),
+          Js.bool(false),
+        );
```

Comment:
I would love if I could just call inject here but I don't think that works with incr_dom's model.